### ISSUE TITLE
Ray traversal mechanism

### ISSUE THREAD

**Issue by deltasquad451** [starandserpent/Terra#69]
Ray casting mechanism for octree traversal and chunk traversal.

- For chunk traversal bresenham's line algorithm would need to be used

- For octree traversal, it would need to be visited in front-to-back order, depth-first

**Comment by Pilvinen** [comment]
Is there a reason why you suggested Bresenham specifically?

https://www.gamedev.net/blogs/entry/2265248-voxel-traversal-algorithm-ray-casting/

**Comment by Pilvinen** [comment]
I have to add that you didn't really specify what you had planned this for beyond the "octree and chunk traversal".

If the indented use is for occlusion queries, as I suspect, then this is probably a wrong approach and we need to discuss about that.

However for uses like block picking by using the world data and camera location and calculating it from there (as opposed to using the mesh as we were doing in the past) this could be excellent.

**Comment by deltasquad451** [comment]
Yes this is for things like physics and any other game logic related actions
